File: 0011_ContainerWithMostWater/solution.c
int min(int* a, int* b){
  return *a < *b ? *a : *b;
}

int maxArea(int* height, int heightsize) {
  int left = 0;
  int right = heightsize - 1;
  
  int ans = 0;
  int newsum = -999;

  while(left < right){
    newsum = min(&height[left], &height[right]) * (right-left);

    if(newsum >= ans){
      ans = newsum;
    }

    (height[left] < height[right]) ? left++ : right--;
  }

  return ans;
}

File: 0011_ContainerWithMostWater/solution.go
// Time: 53ms (93.94%) | Memory: 7.75 (68.98%)
func min(a *int, b *int) int{
  if(*a < *b){
    return *a  
  } else{
    return *b
  }
}
func maxArea(height []int) int {
  size := len(height)    

  left, right := 0, size-1
  ans := 0
  for left < right {
    
    if newSum := min(&height[left], &height[right]) * (right-left); newSum >= ans{
      ans = newSum
    }

    if height[left] < height[right]{
      left++
    } else{
      right--
    }
  }

  return ans
}
